src/components/ResourceBrowser/ResourceList/NodeActionsMenu.tsx

```diff
@@ -61,16 +70,20 @@ export default function NodeActionsMenu({
 
     const handleOpenTerminalAction = () => {
         if (isAuthorized()) {
-            openTerminal(nodeData)
+            const queryParams = new URLSearchParams(location.search)
+            queryParams.set('node', name)
+            history.push(
+                `${location.pathname.split('/').slice(0, -2).join('/')}/${AppDetailsTabs.terminal}/${K8S_EMPTY_GROUP}?${queryParams.toString()}`,
```

Comment:
Suggestion: Can we get base url till K8S_EMPTY_GROUP from a util and re-use that in getTabsBasedOnRole?

Reply:
Little ROI right now